src/edata.cpp

```diff
@@ -89,40 +75,209 @@ ExpData::ExpData(ReturnData const& rdata, std::vector<realtype> sigma_y, std::ve
         for (int ie = 0; ie < nmaxevent; ++ie) {
-            mz.at(iz + rdata.nztrue * ie) = rdata.z.at(iz + rdata.nz * ie) + e(gen);
-            sigmay.at(iz + rdata.nztrue * ie) = sigma_z[iz];
+            observedEvents.at(iz + rdata.nztrue * ie) = rdata.z.at(iz + rdata.nz * ie) + e(gen);
+            observedDataStdDev.at(iz + rdata.nztrue * ie) = sigma_z[iz];
         }
     }
 }
 
-void ExpData::setTimepoints(const double *timepoints) {
-    for (int it = 0; it < nt; ++it) {
-        ts.at(it) = static_cast<const realtype>(timepoints[it]);
+void ExpData::setTimepoints(const std::vector<realtype> &ts) {
+    this->ts = ts;
+    observedData.clear();
+    observedData.resize(nt()*nytrue, getNaN());
+    observedDataStdDev.clear();
+    observedDataStdDev.resize(nt()*nytrue, getNaN());
+}
+    
+std::vector<realtype> ExpData::getTimepoints() const {
+    return ts;
+}
+
+const int ExpData::nt() const {
+    return ts.size();
+}
+    
+realtype ExpData::getTimepoint(int it) const {
+    return ts.at(it);
+}
+    
+void ExpData::setObservedData(const std::vector<realtype> &observedData) {
+    if (observedData.size() == (unsigned) nt()*nytrue) {
+        this->observedData = observedData;
+    } else {
+        if (observedData.size() == 0) {
+            this->observedData.clear();
+        } else {
+            throw AmiException("Input observedData did not match dimensions nt (%i) x nytrue (%i), was %i", nt(), nytrue, observedData.size());
+        }
     }
 }
     
-void ExpData::setObservedData(const double *observedData) {
-    for (int imy = 0; imy < nytrue * nt; ++imy) {
-        my.at(imy) = static_cast<const realtype>(observedData[imy]);
+void ExpData::setObservedData(const std::vector<realtype> &observedData, int iy) {
+    if (observedData.size() == (unsigned) nt()) {
+        for (int it = 0; it < nt(); ++it)
+            this->observedData.at(iy + it*nytrue) = observedData.at(iy);
+    } else {
+        throw AmiException("Input observedData did not match dimensions nt (%i), was %i", nt(), observedData.size());
     }
 }
+    
+bool ExpData::isSetObservedData(int it, int iy) const {
+    if (!observedData.empty()) {
+        return !isNaN(observedData.at(it * nytrue + iy));
+    } else {
+        return false;
+    }
+}
+    
+std::vector<realtype> ExpData::getObservedData() const {
+    return observedData;
+}
+    
+const realtype *ExpData::getObservedDataPtr(int it) const {
+    if (!observedData.empty())
+        return &observedData.at(it*nytrue);
+    else
+        return nullptr;
+}
 
-void ExpData::setObservedDataStdDev(const double *observedDataStdDev) {
-    for (int imy = 0; imy < nytrue * nt; ++imy) {
-        sigmay.at(imy) = static_cast<const realtype>(observedDataStdDev[imy]);
+void ExpData::setObservedDataStdDev(const std::vector<realtype> &observedDataStdDev) {
+    if (observedDataStdDev.size() == (unsigned) nt()*nytrue) {
+        this->observedDataStdDev = observedDataStdDev;
+    } else {
```

Comment:
Avoid unnecessary indentation, use `else if`

Reply:
`else if` would mean that `observedData.size() == (unsigned) nt()*nytrue` will not be checked before 
```
        if (observedData.empty()) {
            this->observedData.clear();
        } else {
            throw AmiException("Input observedData did not match dimensions nt (%i) x nytrue (%i), was %i", nt(), nytrue, observedData.size());
        }
```
is executed. Hence we would first call `this->observedData = std::move(observedData);` and then throw `throw AmiException("Input observedData did not match dimensions nt (%i) x nytrue (%i), was %i", nt(), nytrue, observedData.size());` in most scenarios, which is definitely not what we want.

Reply:
Not sure I get that.
This
```
if(a) {
    A
} else {
    if(b) {
        B
    } else {
        C
    }
}
```
is identical with
```
if(a) {
    A
} else if(b) {
    B
} else {
    C
}
```
except for verbosity...

Reply:
oh, misunderstood a stackoverflow post then :(